x-pack/plugin/stack/src/main/java/org/elasticsearch/xpack/stack/StackTemplateRegistry.java

```diff
@@ -249,10 +255,97 @@ protected List<LifecyclePolicy> getLifecyclePolicies() {
             }
         }
         COMPONENT_TEMPLATE_CONFIGS = Map.copyOf(componentTemplates);
+
+        final Map<String, ComponentTemplate> updatedComponentTemplates = new HashMap<>();
```

Comment:
Two static maps will take up resources no matter if the setting is enabled or not...the way it was before the second map would be created only if the setting is enabled...without any overhead for release builds where the if statement will go through the `else` and return the only one existing static map.